src/exports/la-utils/la-transactions/handlers/sponsoredGas/sponsoredGasContractCall.ts

```diff
@@ -152,37 +154,44 @@ export const sponsoredGasContractCall = async ({
 
   // send the user operation with EIP-7702 delegation in a runOnce
   // so that we don't submit it more than once
-  const uoHash = await Lit.Actions.runOnce(
+  const sendWithAlchemyResult = await Lit.Actions.runOnce(
     {
       waitForResponse: true,
       name: "sendWithAlchemy",
     },
     async () => {
       try {
         // Send the user operation with EIP-7702 delegation
-        const userOpResult = await smartAccountClient.sendRawUserOperation(
+        const userOpHash = await smartAccountClient.sendRawUserOperation(
           signedUserOperation,
           entryPoint.address
         );
 
         console.log(
           `[@lit-protocol/vincent-tool-morpho/executeOperationWithGasSponsorship] User operation sent`,
-          { userOpHash: userOpResult }
+          { userOpHash }
         );
 
-        return userOpResult;
+        const bundleHash = eip7702WaitForBundleTx ? await smartAccountClient.waitForUserOperationTransaction({
+          hash: userOpHash,
+        }) : undefined;
+
+        return JSON.stringify({ bundleHash, userOpHash });
       } catch (e: any) {
         console.log("Failed to send user operation, error below");
         console.log(e);
         console.log(e.stack);
-        return "";
+        // Failure could be just timeout waiting for the userOp to be mined, so we return whatever we have anyway
+        return JSON.stringify({ bundleHash, userOpHash });
       }
     }
   );
 
-  if (uoHash === "") {
+  const { bundleHash, userOpHash } = JSON.parse(sendWithAlchemyResult);
+
+  if (userOpHash === "") {
     throw new Error("Failed to send user operation");
   }
 
-  return uoHash;
+  return eip7702WaitForBundleTx ? userOpHash : { bundleHash, userOpHash };
```

Comment:
Aweful return just to keep current return type when new param is not passed and avoid this being a breaking change